controllers/routemonitor/routemonitor_controller_supplement.go

```diff
@@ -168,16 +167,203 @@ func (r *RouteMonitorReconciler) CreateBlackBoxExporterService(ctx context.Conte
 	}
 	return nil
 }
-func (r *RouteMonitorReconciler) CreateBlackBoxExporterResources(ctx context.Context) error {
-	if err := r.CreateBlackBoxExporterDeployment(ctx); err != nil {
+
+func (r *RouteMonitorReconciler) CreateServiceMonitorResource(ctx context.Context, routeMonitor *v1alpha1.RouteMonitor) (*ctrl.Result, error) {
+	// Was the RouteURL populated by a previous step?
+	if routeMonitor.Status.RouteURL == "" {
+		return nil, customerrors.NoHost
+	}
+
+	if !r.HasFinalizer(routeMonitor) {
+		// If the routeMonitor doesn't have a finalizer, add it
+		utilfinalizer.Add(routeMonitor, FinalizerKey)
+		if err := r.Update(ctx, routeMonitor); err != nil {
+			return nil, err
+		}
+		// After any modification we need to requeue to prevent two threads working on the same code
+		return &ctrl.Result{Requeue: true}, nil
+	}
+
+	namespacedName := r.templateForServiceMonitorName(routeMonitor)
+
+	resource := &monitoringv1.ServiceMonitor{}
+	populationFunc := func() *monitoringv1.ServiceMonitor {
+		return r.templateForServiceMonitorResource(routeMonitor, namespacedName.Name)
+	}
+
+	// Does the resource already exist?
+	if err := r.Get(ctx, namespacedName, resource); err != nil {
+		// If this is an unknown error
+		if !k8serrors.IsNotFound(err) {
+			// return unexpectedly
+			return nil, err
+		}
+		// populate the resource with the template
+		resource := populationFunc()
+		// and create it
+		err = r.Create(ctx, resource)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return nil, nil
+}
+
+/*
+
+This next block is for deletion of the RouteMonitor
+
+*/
+
+func (r *RouteMonitorReconciler) ShouldDeleteBlackBoxExporterResources(ctx context.Context, routeMonitor *v1alpha1.RouteMonitor) (bool, error) {
+	// if a delete has not been requested then there is at least one resource using the BlackBoxExporter
+	if !r.WasDeleteRequested(routeMonitor) {
+		return false, nil
+	}
+
+	routeMonitors := &v1alpha1.RouteMonitorList{}
+	if err := r.List(ctx, routeMonitors); err != nil {
+		return false, err
+	}
+
+	amountOfRouteMonitors := len(routeMonitors.Items)
+	if amountOfRouteMonitors == 0 {
+		err := errors.New("Internal Fault: Cannot be in reconcile loop and have not RouteMonitors on cluster")
+		// the response is set to true as this technically a case where we should delete, but as it's not logical that this will happen, returning error
+		return true, err
+	}
+
+	// If more than one resource exists, and the deletion was requsted there are stil at least one resource using the BlackBoxExporter
+	if amountOfRouteMonitors > 1 {
```

Comment:
What if the last two `RouteMonitor`s are deleted simultaneously?

Reply:
You are right, but I didn't think of a check to fix this so I left this question open